tensorflow/python/ops/image_ops_impl.py

```diff
@@ -3541,3 +3542,63 @@ def combined_non_max_suppression(boxes,
     return gen_image_ops.combined_non_max_suppression(
         boxes, scores, max_output_size_per_class, max_total_size, iou_threshold,
         score_threshold, pad_per_class)
+  
+  
+@tf_export('image.median_filtering_2D')
+def median_filtering_2D(tf_img,filter_shapex=3,filter_shapey=3):
+    # This methods takes both 2D Tensor as well as 3D Tensor Images
+    # Other than Tensor it takes optional parameter filter_Size
+    # Default Filter Size = 3 , 3
+    # This Median Filtering is done by using 2D filters of user's choice
+    # Filter_size should be odd
+    # This method takes both kind of images where pixel values lie between 0 to 255 and where it lies between 0.0 and 1.0
+    
+    try:
+        m,no = int(tf_img.shape[0]),int(tf_img.shape[1])
+    except:
+        raise Exception("Input Tensor is not an image")
+    try :
+        ch = int(tf_img.shape[2])
+    except:
+        ch = 1
+        tf_img = array_ops.reshape(tf_img, [m,no,ch])
+    if m < filter_shapex or no < filter_shapey:
+        raise Exception('No of Pixels in the image should be more than the filter size')
+    if filter_shapex % 2 == 0 or filter_shapey % 2 == 0:
+        raise Exception("Filter size should be odd")
+    sess = session.InteractiveSession()
+    tf_img = tf_img.eval()
```

Comment:
So this is not actually building a tensorflow graph to do anything useful but instead it's escaping to numpy and doing the computation there.

You can achieve this as part of a tf graph by using tf.py_func / tf.numpy_function, but I encourage you instead to try to rewrite this in terms of graph operations.